r2-core/src/main/java/com/linkedin/r2/filter/transport/ServerRetryFilter.java

```diff
@@ -115,16 +115,18 @@ public void onStreamError(Throwable ex,
     {
       if (cause instanceof RetriableRequestException)
```

Comment:
This made me feel that we should use 2 different exceptions on the server side and client side. If they are 2 different exceptions, we won't have issue with a client passing the exception to its caller again. We should always let server construct a new retry request from the overload shedder, there shouldn't be much chance that a server can get exactly the same exception converted from its downstream. But since we've been using this, I agree your solution also works.